networking-calico/networking_calico/agent/dhcp_agent.py

```diff
@@ -457,10 +471,12 @@ def _update_dnsmasq(self, network_id):
         ports_needed.sort(key=lambda port: port.id)
 
         # Compare that against what we've last asked Dnsmasq to handle.
-        if ports_needed != self._last_dnsmasq_ports.get(network_id):
+        last_ports = self._last_dnsmasq_ports.get(network_id)
+        mtu_changing = self._last_dnsmasq_mtu.get(network_id) != net.mtu
```

Comment:
If the network is initially configured with 9k MTU, and then dropped, this may not trigger... because the `defaultdict` assumes that the standard Neutron MTU of 1500 is used? 🤔

Reply:
Sorry, I don't understand the concern here.  Please explain further.

Reply:
Separately, as this is the only place that accesses `self._last_dnsmasq_mtu`, I'd be inclined to make it a normal dict and use `self._last_dnsmasq_mtu.get(network_id, neutron_constants.DEFAULT_NETWORK_MTU)` here.

Reply:
I've overthought this and had put in the `defaultdict` while working on a way to ensure that `dnsmasq` is restarted for any MTU change, but yes -- now realizing that none of it was probably needed. Let me do some quick testing and I'll make those changes.